frame/uniques/src/functions.rs

```diff
@@ -117,6 +117,7 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 			Attribute::<T, I>::remove_prefix((&collection,), None);
 			CollectionAccount::<T, I>::remove(&collection_details.owner, &collection);
 			T::Currency::unreserve(&collection_details.owner, collection_details.total_deposit);
```

Comment:
I think it would make sense to have the collections ID not be user controlled, but have it be an ever increasing integer.  
That way the ID of a unique collection is itself *unique*.

Reply:
Yeah, I completely agree with you. In V2 I've made the ids auto-generated, so you won't be able to get the same ID twice.
At the same time, the main idea behind the current IDs model is that you can batch your call and execute different actions on your collection since you know its ID.